### PR TITLE
Make StaticPage component for Static pages

### LINKED ISSUE #3410: Refactor static pages components
From https://github.com/mozilla/addons-frontend/pull/3207#discussion_r142559054 as mentioned by @tofumatt:

> I'd rather us have a StaticPage component that included this CSS and a bit of the boilerplate Card stuff rather than importing CSS from other components. We've done stuff like this in the past, I think with error pages for instance, but we've since got out the habit of doing it and I think just having a StaticPage component would be better. We tend not to import other component's CSS. Something like:

```JS
// amo/components/StaticPage/index.js
import 'amo/components/StaticPages/styles.scss';

export default class StaticPage extends React.Component { [...] };

// src/amo/components/AboutPage/index.js
import StaticPage from 'amo/components/StaticPage';


export default class AboutPage extends React.Component {
  [...]
  render() {
    return <StaticPage title="About this site">{CONTENT}</StaticPage>
  }
};
```
> But that's a later optimisation. We should just file a bug for it to remind us to tidy up the static pages unless it's super-easy to convert.

### PR DESCRIPTION
**Pull request by xu3u4**
Fixes #3410

Create `StaticPage` component which contains css and components for all static pages.

### CODE REVIEW

File: src/amo/components/StaticPage/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <Card className="StaticPage" header={title}>
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+
+      <HeadMetaTags description={metaDescription} title={title} />
+
+      <HeadLinks />
+
+      <div className="StaticPageWrapper">{children}</div>
```
Comment (willdurand) [inline src/amo/components/StaticPage/index.js]: Could you rename the class name to `StaticPage-content` please?

File: src/amo/pages/StaticPages/ReviewGuide/index.js [hunk 2]
```diff
@@ -1,42 +1,29 @@
...
   i18n: I18nType,
 |};
 
 export class ReviewGuideBase extends React.Component<Props> {
   render() {
     const { i18n } = this.props;
-
-    const title = i18n.gettext('Review Guidelines');
+    const metaDescription = i18n.gettext(`
+      Guidelines, tips, and Frequently Asked Questions to leave a review for the extensions and
```
Comment (willdurand) [inline src/amo/pages/StaticPages/ReviewGuide/index.js]: Why not inlining this content in the component below? (like for the `About` component)

File: tests/unit/amo/components/TestStaticPage.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import Card from 'ui/components/Card';
+
+describe(__filename, () => {
+  const title = 'Static page title';
+  const metaDescription = 'Static page HeadMetaTag description';
+  const children = <div>Static Page Content</div>;
+  const render = () => {
+    const allProps = { title, metaDescription, children };
+
+    return shallow(<StaticPage {...allProps} />);
```
Comment (willdurand) [inline tests/unit/amo/components/TestStaticPage.js]: I suggest putting the variables/default values in the function signature:

```js
const render = ({ title = 'some title', metaDescription = 'meta desc', children = <div /> }) => {
```

Comment (willdurand) [reply tests/unit/amo/components/TestStaticPage.js]: Also I think that we should put `children` like this:

```js
<StaticPage>{children}</StaticPage>
```

**Comment by xu3u4** [comment]
@willdurand it is updated! 
Please check.

**Comment by willdurand** [comment]
Thanks @xu3u4!

**Comment by xu3u4** [comment]
@willdurand thank you for helping me fix it.

**Pull request merged**